sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ArithmeticExpressionSuite.scala

```diff
@@ -576,4 +577,41 @@ class ArithmeticExpressionSuite extends SparkFunSuite with ExpressionEvalHelper
       }
     }
   }
+
+  test("SPARK-34677: exact add and subtract of day-time and year-month intervals") {
+    checkExceptionInExpression[ArithmeticException](
+      UnaryMinus(Literal.create(Period.ofMonths(Int.MinValue), YearMonthIntervalType)),
+      "overflow")
+    Seq(true, false).foreach { failOnError =>
```

Comment:
`UnaryMinus` has the `failOnError` flag as well. Shall we move it inside as well?